package/MDAnalysis/core/groups.py

```diff
@@ -1641,12 +1644,45 @@ def unwrap(self, compound='fragments', reference='com', inplace=True):
                     refpos = refpos.astype(np.float32, copy=False)
                     target = distances.apply_PBC(refpos, self.dimensions)
                     positions[mask] += target - refpos
+
+
         if inplace:
             unique_atoms.positions = positions
         if not atoms.isunique:
             positions = positions[atoms._unique_restore_mask]
         return positions
 
+    def choose_periodic_image(self, reference_point, centre='com', inplace=True):
+        atoms = self.atoms
+
+        unique_atoms = atoms.unique
```

Comment:
`unique_atoms = self.atoms.unique`?

Reply:
No, `atoms` is needed further below for `atoms._unique_restore_mask`